cmd/agent/app/dogstatsd_replay.go

```diff
@@ -168,22 +176,27 @@ func dogstatsdReplay() error {
 	// wait for go routine to start processing...
 	time.Sleep(time.Second)
 
-replay:
-	for {
-		select {
-		case msg := <-reader.Traffic:
-			// The cadence is enforced by the reader. The reader will only write to
-			// the traffic channel when it estimates the payload should be submitted.
-			n, oobn, err := conn.(*net.UnixConn).WriteMsgUnix(
-				msg.Payload[:msg.PayloadSize], replay.GetUcredsForPid(msg.Pid), addr)
-			if err != nil {
-				return err
+iterate:
+	for i := 0; i < dsdReplayIterations || dsdReplayIterations == 0; i++ {
+
+	replay:
+		for {
+			select {
+			case msg := <-reader.Traffic:
+				// The cadence is enforced by the reader. The reader will only write to
+				// the traffic channel when it estimates the payload should be submitted.
+				n, oobn, err := conn.(*net.UnixConn).WriteMsgUnix(
+					msg.Payload[:msg.PayloadSize], replay.GetUcredsForPid(msg.Pid), addr)
+				if err != nil {
+					return err
+				}
+				fmt.Printf("Sent Payload: %d bytes, and OOB: %d bytes\n", n, oobn)
+			case <-reader.Done:
+				reader.Seek(0)
+				break replay
+			case <-done:
+				break iterate
```

Comment:
nit: I don't have a strong opinion on this but labels/`breaks` are hard to follow usually so I would generally feel like this code would be a bit easier to maintain if we can extract this inner loop into a function and remove the dual breaks.